CMakeLists.txt

```
@@ -69,6 +69,8 @@ set(IREE_RELEASE_VERSION "0.1a1" CACHE STRING "Version to embed in distributed p
 set(IREE_RELEASE_REVISION "HEAD" CACHE STRING "Version control revision information to embed in distributed packages")
 option(IREE_EMBED_RELEASE_INFO "Embed the IREE version information in built artifacts." OFF)
 
+set(IREE_HOST_BINARY_ROOT "" CACHE STRING "Path to installed IREE tools, to use instead of building them from source")
```

How about we introduce a new variable that actually does the thing we want, rather than trying to change the meaning of `IREE_HOST_BINARY_ROOT` by removing the implicit "/bin" suffix:

```suggestion
set(IREE_HOST_BINARY_ROOT "" CACHE STRING "Path to installed IREE tools, to use instead of building them from source")

set(IREE_HOST_BIN_DIR_DEFAULT "")
if(IREE_HOST_BINARY_ROOT)
  message(WARNING "IREE_HOST_BINARY_ROOT is deprecated. Use IREE_HOST_BIN_DIR"
                  " pointing directly to the directory containing binaries"
                  " instead.")
  set(IREE_HOST_BIN_DIR_DEFAULT "${IREE_HOST_BINARY_ROOT}/bin")
endif()
set(IREE_HOST_BIN_DIR "${IREE_HOST_BIN_DIR_DEFAULT}" CACHE STRING "Path to directory containing IREE binary tools to use instead of building them from source.")
```

---

I like this idea, though I'm wondering about commit sequencing. Maybe I should
1) revert the changes in this PR that define the `IREE_HOST_BINARY_ROOT` variable and keep checking if `IREE_HOST_BINARY_ROOT` is defined
2) send a second PR (or second commit in this PR) that does as you suggest with a new variable, initialized from the old variable
3) send a third PR removing the old variable after some time

---

I reverted the variable changes in this PR. I'll leave that (and your suggested change) to a follow-up.